backend/pkg/models/v1beta/job_offers.go

```diff
@@ -9,41 +9,51 @@ import (
 
 // JobOffer defines the job_offers structure
 type JobOffer struct {
-	ID        int64     `json:"id" gorm:"column:id"`
+	ID        int64     `json:"id" gorm:"column:id;primaryKey;autoIncrement:true"`
 	CreatedAt time.Time `json:"createdat" gorm:"column:createdat"`
 	UpdatedAt time.Time `json:"updatedat" gorm:"column:updatedat"`
 
-	Email             string `json:"email" gorm:"column:email"`
 	CompanyName       string `json:"company_name" gorm:"column:company_name"`
+	CompanyEmail      string `json:"company_email" gorm:"column:company_email"`
 	TitleID           int64  `json:"title" gorm:"column:title_id"`
 	IsRemote          bool   `json:"is_remote" gorm:"column:is_remote"`
+	Location          string `json:"location" gorm:"column:location"`
+	Department        string `json:"department" gorm:"column:department"`
+	SalaryRangeMin    int64  `json:"salary_range_min" gorm:"column:salary_range_min"`
+	SalaryRangeMax    int64  `json:"salary_range_max" gorm:"column:salary_range_max"`
 	Description       string `json:"description" gorm:"column:description"`
+	Benefits          string `json:"benefits" gorm:"column:benefits"`
 	HowToApply        string `json:"how_to_apply" gorm:"column:how_to_apply"`
 	ApplyUrl          string `json:"apply_url" gorm:"column:apply_url"`
 	ApplyEmailAddress string `json:"apply_email_address" gorm:"column:apply_email_address"`
+	ApplyPhoneNumber  string `json:"apply_phone_number" gorm:"column:apply_phone_number"`
+	Tags              string `json:"tags" gorm:"column:tags"`
 }
 
 // OfferPostQuery defines the body object used to create a new jb offer on a POST query
 type OfferPostQuery struct {
-	Email             string `json:"email"`
 	CompanyName       string `json:"company_name"`
+	CompanyEmail      string `json:"company_email"`
 	JobTitle          string `json:"job_title"`
 	IsRemote          bool   `json:"is_remote"`
+	Location          string `json:"location"`
+	Department        string `json:"department"`
+	SalaryRangeMin    int64  `json:"salary_range_min"`
+	SalaryRangeMax    int64  `json:"salary_range_max"`
 	Description       string `json:"description"`
+	Benefits          string `json:"benefits"`
 	HowToApply        string `json:"how_to_apply"`
 	ApplyUrl          string `json:"apply_url"`
 	ApplyEmailAddress string `json:"apply_email_address"`
+	ApplyPhoneNumber  string `json:"apply_phone_number"`
+	Tags              string `json:"tags"`
 }
 
-// Validate check if the mandatory fields are filled
-func (r OfferPostQuery) Validate() error {
-	if strings.TrimSpace(r.Email) == "" {
-		return errors.New("email is mandatory")
-	}
-
+// Validate check if the mandatory fields are filled, and make some changes in the tags field
+func (r *OfferPostQuery) Validate() error {
```

Comment:
@rakici  since that the e2e tests test this function who is used in the post job offers handlers, do we still need to add some test on the golang side ?

Reply:
I wrote tests for this